docs/auth/codegarten-api.md

```diff
@@ -0,0 +1,17 @@
+# CodeGarten API Access
+
+## Required Information
+In order to manage access to the CodeGarten API, it's required to store the following information:
+* CodeGarten Client - A client that's registered to use the API. This client will have:
+    * Client ID - Used to identify what client is requesting an authorization code/access token
+    * Client Name - User friendly way of identifying the client
+    * Client Secret - Used when exchanging an authorization code for an access token
+    * Client Redirect URI - The URI that will receive the authorization code
+
+* CodeGarten API Access Token - Used to access the API endpoints that require authentication. It's associated with a user and a client
+* CodeGarten API Access Token Expiration Date - Used to determine when to request a new Access Token
+
+## Obtaining an Access Token
+In order to obtain a CodeGarten API Access Token, it's required to authenticate via GitHub. This process is described in the [corresponding documentation](github-api.md#user-access-token). It's also required to specify which client is requesting an access token by sending the client ID through the query string.

```

Comment:
"sending the client ID through the query string" - this follows the OAuth protocol, right?